### PR TITLE
Remove KVault dependency that contains androidx.

### PR DESCRIPTION
**Pull request by AfanasievAnton**

### CODE REVIEW

File: transport/src/iosMain/kotlin/com/genesys/cloud/messenger/transport/util/TokenStoreImpl.kt [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
-import com.liftric.kvault.KVault
 import platform.Foundation.NSUUID
 
 internal class TokenStoreImpl(storeKey: String) : TokenStore {
 
-    private val store = KVault(storeKey)
+    private val store = Vault(storeKey)
     override val token: String
         get() = store.string(TOKEN_KEY) ?: NSUUID().UUIDString().also {
             store.set(TOKEN_KEY, it)
```
Comment (crumpf) [inline transport/src/iosMain/kotlin/com/genesys/cloud/messenger/transport/util/TokenStoreImpl.kt:10]: It looks like this `set()` returns a bool for success/failure. If the set fails would it be worthwhile to log the failure?

**Review by bjdupuis — APPROVED** [review]

File: transport/src/iosMain/kotlin/com/genesys/cloud/messenger/transport/util/TokenStoreImpl.kt (hunk 1 above)
Comment (AfanasievAnton) [reply transport/src/iosMain/kotlin/com/genesys/cloud/messenger/transport/util/TokenStoreImpl.kt]: PR updated.

File: transport/src/iosMain/kotlin/com/genesys/cloud/messenger/transport/util/MobileMessenger.kt [hunk 2]
```diff
@@ -19,7 +19,7 @@ object MobileMessenger {
         val log = Log(configuration.logging, LogTag.MESSAGING_CLIENT)
         val api = WebMessagingApi(log.withTag(LogTag.API), configuration)
         val webSocket = PlatformSocket(log.withTag(LogTag.WEBSOCKET), configuration, 300000)
-        val token = TokenStoreImpl(configuration.tokenStoreKey).token
+        val token = TokenStoreImpl("configuration.tokenStoreKey23", log.withTag(LogTag.TOKEN_STORE)).token
```
Comment (crumpf) [inline transport/src/iosMain/kotlin/com/genesys/cloud/messenger/transport/util/MobileMessenger.kt:22]: It looks like `storeKey` param is using a string literal `"configuration.tokenStoreKey23"` now

Comment (AfanasievAnton) [reply transport/src/iosMain/kotlin/com/genesys/cloud/messenger/transport/util/MobileMessenger.kt:22]: OMG!!! What a great catch. I do not believe that I forgot to undo that change (did it to check that log.e actually work)

**Review by crumpf — APPROVED** [review]

**Pull request merged**